course_catalog/models.py

```diff
@@ -72,45 +67,70 @@ class Course(LearningResource):
     image_src = models.URLField(max_length=400, null=True, blank=True)
     image_description = models.CharField(max_length=1024, null=True, blank=True)
     last_modified = models.DateTimeField(null=True, blank=True)
-    raw_json = JSONField(null=True, blank=True)
+
+    language = models.CharField(max_length=128, null=True, blank=True)
     featured = models.BooleanField(default=False)
     published = models.BooleanField(default=True)
     availability = models.CharField(max_length=128, null=True, blank=True)
     url = models.URLField(null=True, max_length=2048)
+
+    learning_resource_type = models.CharField(
+        max_length=20, default=ResourceType.course.value
+    )
+
+    class Meta:
+        abstract = True
+
+
+class Course(AbstractCourse):
+    """
+    Course model for courses on all platforms
+    """
+
+    course_id = models.CharField(max_length=128, unique=True)
+
+    level = models.CharField(max_length=128, null=True, blank=True)
+    semester = models.CharField(max_length=20, null=True, blank=True)
+    platform = models.CharField(max_length=128)
+
+    raw_json = JSONField(null=True, blank=True)
+
     program_type = models.CharField(max_length=32, null=True, blank=True)
     program_name = models.CharField(max_length=256, null=True, blank=True)
+
     instructors = models.ManyToManyField(
         CourseInstructor, blank=True, related_name="courses"
     )
-    learning_resource_type = models.CharField(
-        max_length=20, default=ResourceType.course.value
+    prices = models.ManyToManyField(CoursePrice, blank=True)
+
+
+class CourseRun(AbstractCourse):
+    """
+    Model for course runs
+    """
+
+    course_run_id = models.CharField(max_length=128, unique=True)
+    semester = models.CharField(max_length=20, null=True, blank=True)
+    platform = models.CharField(max_length=128)
```

Comment:
`platform` should be the same for every run in a `Course`, so it might not be needed here?